### DISCUSSION TITLE
imagick driver: 'format' method issue

### DISCUSSION THREAD

**Discussion by lightsuner** [spatie/image#228] [General]
Hello Spatie,

Thanks for the library!

-------
Issue
-------

I've come across different behaviours of `gd` and `imagick` drivers.

- My goal was to resize JPEG and save it as `avif`.
- Also, I wanted to use php functionality for temporary files - `tmpfile()`.

`tmpfile()` - creates a file without an extension, like: `/tmp/phpFx0513a`.

So here is a code example:

```php
$driver = 'imagick'; // or 'gd'
$image = SpatieImage::useImageDriver($driver)->loadFile('/path/to/original/image.jpeg');
$image->resize(100, 150);

$tmpFile = tmpfile();
$tmpFilePath = stream_get_meta_data($tmpFile)['uri'];

$image->format('avif')->save($tmpFilePath);
````

- So when the `$driver` is `gd`, everything works as expected.
- When the `$driver` is `imagick`, I receive the following error:
````
  Spatie\Image\Exceptions\UnsupportedImageFormat 

  Unsupported format ``.
````

In my opinion, there is a bug in the `imagick` driver, as it tries to fetch the format from the destination file's extension and ignores the `$image->format('avif')` method.

Thanks.

**Comment by freekmurze** [comment]
Hi, could you submit a PR that fixes this? If possible, include a test.

**Comment by FrozenGod** [comment]
I get the same `Unsupported format ``.` with all WebP images on both imagick and GD. Also spent a whole day trying to stream a webp media with no success. Something is definitely still wrong. I haven't checked if there's a PR but if it's there it's either not working or there is another bug elsewhere.

Ended up using `Storage::disk($media->disk)->get($media->getPathRelativeToRoot());` instead to load the file instead which worked perfectly fine.
